---
title: "easyFulcrum"
author: "Author: Matteo Di Bernardo"
date: "`r format(Sys.time(), '%d %b %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{easyFulcrum}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = F}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

The **easyfulcrum** package is a tool to work with data from Fulcrum exports, Google Sheets containing corresponding genotyping results, BLAST results, and images from collection.

Install and load the package:

```{r setup}
# install.packages("devtools")
# devtools::install_github("AndersenLab/easyfulcrum")
# setwd("~/Desktop")

library(easyfulcrum)
```

As you may have noticed in the previous code chunk, fully running the **easyfulcrum** package relies heavily on the placement of files in appropriate directories for read in into the package pipeline. This makes the the universal building of this vignette impossible as a users personal directory cannot be specified. Therefore, certain lines of code and the running of certain functions is commented out as to not be part of the local vignette build. We will specify when this is the case.

### Directory structure:

The `makeDirStructure` function makes a standardized directory of folders for the `easyfulcrum` run, taking a base directory (`startdir`) and the project name (`projectdirname`) as inputs.<br> 

Every collection project should be contained in its own directory. The directory name should follow the `YearMonthPlace` format used for Fulcrum collection projects, e.g. `2020JanuaryHawaii`.<br>

See the [Nematode Collection Protocol](https://docs.google.com/document/d/1aPgPDYobpxzr0BQYgvUsjPWnAGXZmCSQKQ4sDGBXuYM/edit)  for more details.<br>

As this depends on a users personal subfolder structure, this function is commented out.

```{r warning = F, message = F}
# makeDirStructure(startdir = "~/Desktop/",
#                  projectdirname = "2020JanuaryHawaii")
```

The `data` directory contains the  `raw` and `processed` subdirectories.<br>
    - `raw/fulcrum` holds the `.csv` files exported from Fulcrum and `raw/fulcrum/photos` contains `.jpg` files exported from Fulcrum.<br> -->
    - `raw/sanger` can hold `.ab1`, `.phd.1`, `.scf`, and `.seq` files exported from the sequencing facility. We have not implemented functions for merging Sanger files.<br>
    - `raw/annotate` can hold spatial location files `island.csv`, `location.csv`, and `trail.csv` that the user generates for mapping the collection sites.<br>
    - `processed/fulcrum` and `processed/sanger` directories hold `easyfulcrum` function outputs.<br>
The `reports` directory holds `easyfulcrum` function outputs. These outputs will be generated by the user processing script(s) saved in the `scripts` directory.<br>

In a normal `easyfulcrum` run, the user would add collection `.csv` and `.jpg` files into the appropriate locations after running this function.

### Reading, processing, and joining Fulcrum results:

The first group of functions cleans the results from the Fulcrum `.csv` files.<br>

#### readFulcrum:

`readFulcrum` takes a `dir` argument that specifies the directory to read in Fulcrum `.csv` files. This will be useful throughout the package.

As this depends on a users personal subfolder structure, this function is commented out (`raw_data1` object is automatically loaded as the example raw fulcrum data).

```{r}
dir1 <- "~/Desktop/2020JanuaryHawaii"
# raw_fulc1 <- readFulcrum(dir = dir1)
raw_fulc1 <- easyfulcrum::raw_fulc1
names(raw_fulc1)
```

#### procFulcrum:

`procFulcrum` processes individual data frames and adds flags for unexpected data.

```{r}
proc_fulc1 <- procFulcrum(data = raw_fulc1)
```

#### checkTemperatures:

`checkTemperatures` identifies flags in three temperature variables. Setting the `return_flags` option to `TRUE` will return a list of three data frames that pulls only the rows where each of the three flag types appear. The function automatically prints the rows where the flags exist.

```{r}
flag1.1 <- checkTemperatures(data = proc_fulc1, return_flags = TRUE)
```

`procFulcrum` function assumes that, when `raw_substrate_temperature` or `raw_ambient_temperature` temperatures are above 40 degrees, the temperatures were mistakenly input as Fahrenheit rather than Celsius, and converts these values to Celsius. It will also notice when both `raw_ambient_temperature` and `raw_ambient_humidity` get stuck on the same value for 5 or more measurements in a row. These are the three flags returned above.<br>

#### fixTemperatures: 

`fixTemperatures` takes a) `fulcrum_id`s that need to be reverted *back to their original values* (if readings above 40 degrees were truly in Celsius) for both substrate temperatures (`substrate_temperature_ids`) and ambient temperatures (`ambient_temperature_ids`) as well as b) `fulcrum_id`s for which humidity and temperature readings need to be set to NA due to a stuck measurement device (`ambient_temperature_run_ids`).

```{r}
proc_fulc1_clean <- fixTemperatures(data = proc_fulc1,
                                  substrate_temperature_ids = "a7db618d-44cc-4b4a-bc67-871306029274",
                                  ambient_temperature_ids = "b1f20ae4-c5c2-426f-894a-e1f46c2fa693",
                                  ambient_temperature_run_ids=c("dda77efe-d73c-48e9-aefb-b508e613256b",
                                                                 "93de14a0-40ab-4793-8614-ab1512ab158c"))
```

The flag variables will be maintained, so rerunning `checkTemperatures` on the cleaned processed fulcrum data to ensure that corrections have been implemented as desired.

#### joinFulcrum:

`joinFulcrum` joins the Fulcrum dataframes. The function works to first join the processed nematode field sampling dataframe to the processed nematode isolation dataframe via unique collection labels (`c_label`). Following this join, it selects a “best” photo for each unique c label based on the existence of a matching photo id in the processed nematode field sampling sample photo dataframe, before joining the aggregated data to this dataframe as well. Finally, the complete merge of the dataframes is achieved when the processed nematode isolation s labeled plates is joined to this large dataframe on the basis of isolation id (`s_label`).<br>

```{r}
join_fulc1 <- joinFulcrum(data = proc_fulc1)
```

#### checkJoin:

`checkJoin` conducts 10 different checks for flags in the joined Fulcrum data frame regarding extreme temperatures and altitude values, missing, improper, and/or duplicated c labels, missing, and/or duplicated isolation records corresponding to these c labels, and finally unusual sample photo numbers. The `return_flags` option is the same as in `checkTemperatures`, and the function automatically prints the rows where the flags exist. If desired, the user can manually edit values or correct mistakes in the underlying data based on these flags and quickly re-run the pipeline again.

```{r}
flag1.2 <- checkJoin(data = join_fulc1, return_flags = TRUE)
```

#### annotateFulcrum:

`annnotateFulcrum` adds spatial information to the joined Fulcrum data frame, noting if sample collections were collected on specific islands, trails, and/or locations. Examples islands, trails, and/or locations are automatically loaded with the package, but a user can specify manually made `.csv` files and place them in _data/raw/annotate_, specifying the base directory as `dir` in `annotateFulcrum` will override the example files.<br>

The `hawaii_islands` and `hawaii_locations` dataframes are composed of simple latitude and longitude starts and ends to create a bounding box, and `hawaii_trails` is composed of a character list of geojson polygon points from geojson output of that can be created on a [bounding box online tool](https://boundingbox.klokantech.com/).

```{r message = F}
anno_fulc1 <- annotateFulcrum(data = join_fulc1, dir = NULL)
```

### Reading, processing, and joining genotyping google sheet:

This second group of functions operates to clean the results from the collection specific Google Sheet of genotyping results.<br>

Details on how to create genotyping Google Sheets can be found in the [Nematode Collection Protocol](https://docs.google.com/document/d/1aPgPDYobpxzr0BQYgvUsjPWnAGXZmCSQKQ4sDGBXuYM/edit), look for "wild_isolate_genotyping_template".<br>

#### readGenotypes:

`readGenotypes` reads in genotyping data from a Google Sheet with requisite `gsKey`. The `col_types` variable will specify of what column types the data should be.<br>

For more details on reading in genotyping sheets, look into the [googlesheets4](https://googlesheets4.tidyverse.org/articles/articles/read-sheets.html) package (which underlies this function), as well as further information on how to [specify](https://googlesheets4.tidyverse.org/reference/range_read.html#column-specification) the `col_types` if needed.<br>

As this depends on a users connection to the appropriate google doc, this function is commented out (`raw_geno1` object is automatically loaded as the example raw geno data).

```{r}
# raw_geno1 <- readGenotypes(gsKey = c("1_6u4sk_Zj-Hm5d_058Lg8WYWLe7BZHGTWxXcH6EsDUI"))
raw_geno1 <- easyfulcrum::raw_geno1
head(raw_geno1)
```

#### checkGenotypes:

`checkGenotypes` both processes the genotyping data (adds flags) and returns info on those flags if desired, regarding isolations missing, improper, and/or duplicated s labels, and other checks such as if species ID, strain name, proliferation label and its2 genotype are missing and expected based on data in the genotyping sheet. Setting the `return_geno` option to `TRUE` and the `return_flags` option to `FALSE` will return the processed genotyping data and print information on the flags. Setting the `return_geno` option to `FALSE` and the `return_flags` option to `TRUE` will return a list of data frames that detail the rows where the flags appear. Both of these cannot be `TRUE` at the same time.

```{r}
proc_geno1 <- checkGenotypes(geno_data = easyfulcrum::raw_geno1, fulc_data = anno_fulc1, 
                                  return_geno = TRUE, return_flags = FALSE)
```

```{r}
flag1.3 <- checkGenotypes(geno_data = raw_geno1, fulc_data = anno_fulc1, 
                          return_geno = FALSE, return_flags = TRUE)
```

#### joinGenoFulc:

`joinGenoFulc` will join the joined Fulcrum data frame with the genotyping information. This function will also save the processed genotyping information in _data/processed/genotypes_ if desired.

```{r}
join_genofulc1 <- joinGenoFulc(geno = proc_geno1, fulc = anno_fulc1, dir = dir1)
```

### Reading, resizing, and joining collection images:

The final function processes and resizes images, adding details to a final dataframe.<br>

```{r message = F, include = F}
#This chunk of code will move the photos required for this trial run into the appropriate folder for raw photos, according to what `dir1` is specified. 
#You can also find these photos for manual download at: https://github.com/AndersenLab/easyfulcrum/tree/master/vignettes/2020JanuaryHawaii_photos

# library(googledrive)
# final_directory <- paste(dir1, "data/raw/fulcrum/photos", sep = "/")
# temp_dir <- tempdir()
# photos <- googledrive::drive_ls(as_id("11T6qzszJ_yK3yRyy4eqBfplZuai20y-g"))
# photos$temp_location <- paste(temp_dir,photos$name, sep = "/")
# photos$final_location <- paste(final_directory,photos$name, sep = "/")
# 
# for(i in 1:nrow(photos)){
#   googledrive::drive_download(as_id(photos$id[i]), 
#                             path =photos$temp_location[i], 
#                             overwrite = TRUE)
#   fs::file_copy(photos$temp_location[i], photos$final_location[i], overwrite = TRUE)
# }
```

#### procPhotos:

`procPhotos` processes the raw sample photos and adds information regarding them to the final data frame of data, which is also saved in _data/processed/fulcrum_. Setting the `CeNDR` option to `TRUE` will place photos of samples meeting CeNDR criteria to an appropriate subfolder.

As this depends on a users personal subfolder structure (`dir1`) and it containing the proper images, this function is commented out (`final_data1` object is automatically loaded as the example final merged data). This also means that the resized images will not visible in the vignette build.

```{r message = FALSE}
# final_data1 <- procPhotos(dir = dir1, data = join_genofulc1,
#                           max_dim = 500, overwrite = T, 
#                           CeNDR = TRUE)
final_data1 <- easyfulcrum::final_data1
head(final_data1)
```

### Generating summary and output files:

We include two functions for generating sample output from `final_data1`. This final dataframe can otherwise be used as needed by the user.

#### makeSpSheet:

`makeSpSheet` generates a species specific `.csv` file for the species of interest (`target_sp`) and writes it to the `/reports` subdirectory. This function simplifies the output of the final dataframe, pulling variables of particular interest for a user specified species of interest. It also  returns a dataframe with flags for these select samples.

```{r}
flag1.4 <- makeSpSheet(data = join_genofulc1,
                    target_sp = "Caenorhabditis Briggsae", dir = dir1)
```

#### sampleReport:

`sampleReport` is a markdown file that will generate a simple description of the data in the project based on `final_data1`. The file will generate output to summarize collection project metadata (such as who conducted the respective processes and on what dates they were completed), to generate summary tables of collection and isolation data, to produce visual maps of where the collections in the project were acquired, and finally to graph basic box and whisker plots of substrate temperature, ambient temperature, humidity, and elevation for the project.

Please feel free to edit this file as you require, the first command will copy the file .Rmd file into `/scripts` sub-directory. We suggest specifying a location for output of the .html file, such as the `/reports` sub-directory.

```{r message = F}
file.copy("../R/sampleReport.Rmd", paste(dir1,"scripts",sep = "/"))

rmarkdown::render(paste(dir1,"scripts","sampleReport.Rmd",sep = "/"),
                  output_dir = paste(dir1,"reports",sep = "/"),
                  output_format	= "html_document")
```
