<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A Tool for Processing and Analyzing Fulcrum Data Generated by The Andersen Lab • easyfulcrum</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="A Tool for Processing and Analyzing Fulcrum Data Generated by The Andersen Lab">
<meta property="og:description" content='This package contains functions to process and analyze data exported from the Fulcrum applications "Nematode Field Sampling" and "Nematode Isolation". These appications are maintained by the Andersen Lab and permission must be granted to export the data. The Nematode Field Sampling app facilitates the sampling of nematodes from nature by offering an easy-to-use interface to record data on substrate, temperature, humidity, and other ecological parameters. The app gives investigators the ability to pair these data with GPS position and photographic evidence at the sampling site. Once nematodes are sampled from nature, the companion application called "Nematode Isolation” is used to record data generated during the isolation process. The primary function of the easyfulcrum package is to match isolation records back to the field sampling records. It also implements diagnostic reporting to review trends in collection data and alert investigators of potential errors within a collection.   '>
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">easyfulcrum</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/easyFulcrum.html">easyFulcrum</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="easyfulcrum" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#easyfulcrum" class="anchor"></a>easyfulcrum</h1></div>
<p>The <code>easyfulcrum</code> package is a tool to work with data from Fulcrum exports, genotyping Google sheets, blast results, and images from collection.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The package is made for specific use with Fulcrum generated collection data. <code>easyfulcrum</code> is specialized for use in the Andersen Lab and, therefore, is not available from CRAN. To install easyfulcrum you will need the <a href="https://github.com/hadley/devtools"><code>devtools</code></a> package. You can install <code>devtools</code> and <code>easyfulcrum</code> using the commands below:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"AndersenLab/easyfulcrum"</span><span class="op">)</span></code></pre></div>
<p>The functionality of the package can be broken down into three main goals:</p>
<ul>
<li><p>Reading raw data from Fulcrum exports, genotyping Google sheets, blast results, and images from collection.</p></li>
<li><p>Joining these data, flagging anomalies, and allowing the user to correct anomalies.</p></li>
<li><p>Exporting corrected data and generating summary reports.</p></li>
</ul>
</div>
<div id="directory-structure" class="section level2">
<h2 class="hasAnchor">
<a href="#directory-structure" class="anchor"></a>Directory structure</h2>
<p>Every collection project should be contained in its own repository. The repository name should follow the <code>YearMonthPlace</code> format used for Fulcrum collection projects, e.g. <code>2020JanuaryHawaii</code>. Given the complexity of the directory creation, we have created a function <code>mkDirStructure</code> that takes two inputs to create an empty substructure for the user to populate, eg:<br></p>
<pre class="{r}"><code><a href="reference/makeDirStructure.html">makeDirStructure(startdir = "~/Desktop/", projectdirname = "2020JanuaryHawaii")</a></code></pre>
<p>The directory structure within the repository is critically important for <code>easyfulcrum</code> functions.<br></p>
<p>The <code>data</code> directory contains the <code>raw</code> and <code>processed</code> subdirectories.<br> - <code>raw/fulcrum</code> holds the <code>.csv</code> files exported from Fulcrum and <code>raw/fulcrum/photos</code> contains <code>.jpg</code> files exported from Fulcrum.<br> - <code>raw/sanger</code> can hold <code>.ab1</code>, <code>.phd.1</code>, <code>.scf</code>, and <code>.seq</code> files exported from the sequencing facility. We have not implemented functionalities for merging sanger files but will work towards this for the next release.<br> - <code>raw/annotate</code> can hold spatial location files <code>island.csv</code>, <code>location.csv</code>, and <code>trail.csv</code> that the user generates for mapping the site of collection.<br> - <code>processed/fulcrum</code> and <code>processed/sanger</code> directories hold <code>easyfulcrum</code> function outputs.<br> The <code>reports</code> directory holds <code>easyfulcrum</code> function outputs. These outputs will be generated by the user processing script(s) saved in the <code>scripts</code> directory.<br></p>
<pre><code>2020JanuaryHawaii/
├── data
│   ├── raw
│       ├── fulcrum
│           ├── nematode_field_sampling.csv
│           ├── nematode_field_sampling_sample_photo.csv
│           ├── nematode_isolation.csv
│           ├── nematode_isolation_s_labeled_plates.csv
│           ├── nematode_isolation_photos.csv
│           ├── photos
│               ├── 0a7a5879-8453-4f20-ab3b-8eabb725d492.jpg
│               ├── 0b16d8c7-3cb3-44a8-8b61-e0789e4062c2.jpg
│               └── ... all collection photos here
│       ├── sanger
│           ├── email@northwestern.edu_01_SEQ1677048A_122319D
│               ├── S-10206_oECA306_A01.ab1
│               ├── S-10206_oECA306_A01.phd.1
│               ├── S-10206_oECA306_A01.scf
│               ├── S-10206_oECA306_A01.seq
│               └── ... more sequences here if present
│           └── ... more sequence folders here if present
│       ├── annotate
│           ├── island.csv
│           ├── location.csv
│           └── trails.csv
│   ├── processed
│       ├── fulcrum
│           ├── empty
│       ├── genotypes
│           ├── empty
│       ├── sanger
│           ├── empty
├── reports
│   ├── empty
├── scripts
│   ├── empty</code></pre>
<p>This directory exhibits the minimal file content and naming for all the <code>easyfulcrum</code> functions to work. The file names in the <code>photos</code> directory should not be altered from the Fulcrum export, the names shown here are examples only. The <code>data/raw/sanger</code> and <code>data/processed/sanger</code> directories are optional. In order to use <code>easyfulcrum</code> blast analysis functions the <code>data/raw/sanger</code> directory must contain <code>.seq</code> and <code>.ab1</code> files exported from the sequencing facility, and the use must develop their own functionalities for integrating this. The file names must contain the S-label and primer used. The folders containing the raw sequence data are optional, they are included in this example because this is the way the data are often exported from the sequencing facility. The <code>data/processed/fulcrum</code>, <code>data/processed/sanger</code>, and <code>reports</code> directories are used to hold <code>easyfulcrum</code> function outputs. The <code>scripts</code> folder will hold some collection processing script(s).</p>
</div>
<div id="project-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#project-workflow" class="anchor"></a>Project workflow</h2>
<p>Nematode collections are performed following a detailed protocol <a href="https://docs.google.com/document/d/1jssQVPFrFsXJiA6Jt7LyEjQRBGfcfGCitP4qPya7VwU/edit">found here.</a> The <code>easyfulcrum</code> package is written to process and analyze the data generated for a specific collection project.<br></p>
<p>The basic workflow is to generate a final processed dataframe that can be used for downstream analysis. This occurs in three major processes, as outlined below:<br></p>
<p><img src=".readmefiles/easyFulcrum_simple.jpg" alt="Project workflow simple"></p>
<p>A detailed description of the 14 <code>easyfulcrum</code> functions is contained in an example walkthrough of the package functionalities, see the vignette for details:<br></p>
<pre class="{r}"><code><a href="https://rdrr.io/r/base/library.html">library(easyfulcrum)
browseVignettes(package = "easyfulcrum")</a></code></pre>
<p>A more complete map of the 14 functions is also shown below:<br></p>
<p><img src=".readmefiles/easyFulcrum.jpg" alt="Project workflow"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Tim Crombie <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5645-4154" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Daniel Cook <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3347-562X" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Tim Crombie, Daniel Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
