<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>easyFulcrum • easyfulcrum</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="easyFulcrum">
<meta property="og:description" content="easyfulcrum">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">easyfulcrum</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/easyFulcrum.html">easyFulcrum</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="easyFulcrum_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>easyFulcrum</h1>
                        <h4 class="author">Author: Matteo Di Bernardo</h4>
            
            <h4 class="date">03 Jun 2021</h4>
      
      
      <div class="hidden name"><code>easyFulcrum.Rmd</code></div>

    </div>

    
    
<p>The <strong>easyfulcrum</strong> package is a tool to work with data from Fulcrum exports, Google Sheets containing corresponding genotyping results, BLAST results, and images from collection.</p>
<p>Install and load the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="co"># devtools::install_github("AndersenLab/easyfulcrum")</span>
<span class="co"># setwd("~/Desktop")</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">easyfulcrum</span><span class="op">)</span></code></pre></div>
<p>As you may have noticed in the previous code chunk, fully running the <strong>easyfulcrum</strong> package relies heavily on the placement of files in appropriate directories for read in into the package pipeline. This makes the the universal building of this vignette impossible as a users personal directory cannot be specified. Therefore, certain lines of code and the running of certain functions is commented out as to not be part of the local vignette build. We will specify when this is the case.</p>
<div id="directory-structure" class="section level3">
<h3 class="hasAnchor">
<a href="#directory-structure" class="anchor"></a>Directory structure:</h3>
<p>The <code>makeDirStructure</code> function makes a standardized directory of folders for the <code>easyfulcrum</code> run, taking a base directory (<code>startdir</code>) and the project name (<code>projectdirname</code>) as inputs.<br></p>
<p>Every collection project should be contained in its own directory. The directory name should follow the <code>YearMonthPlace</code> format used for Fulcrum collection projects, e.g. <code>2020JanuaryHawaii</code>.<br></p>
<p>See the <a href="https://docs.google.com/document/d/1aPgPDYobpxzr0BQYgvUsjPWnAGXZmCSQKQ4sDGBXuYM/edit">Nematode Collection Protocol</a> for more details.<br></p>
<p>As this depends on a users personal subfolder structure, this function is commented out.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># makeDirStructure(startdir = "~/Desktop/",</span>
<span class="co">#                  projectdirname = "2020JanuaryHawaii")</span></code></pre></div>
<p>The <code>data</code> directory contains the <code>raw</code> and <code>processed</code> subdirectories.<br> - <code>raw/fulcrum</code> holds the <code>.csv</code> files exported from Fulcrum and <code>raw/fulcrum/photos</code> contains <code>.jpg</code> files exported from Fulcrum.<br> –&gt; - <code>raw/sanger</code> can hold <code>.ab1</code>, <code>.phd.1</code>, <code>.scf</code>, and <code>.seq</code> files exported from the sequencing facility. We have not implemented functions for merging Sanger files.<br> - <code>raw/annotate</code> can hold spatial location files <code>island.csv</code>, <code>location.csv</code>, and <code>trail.csv</code> that the user generates for mapping the collection sites.<br> - <code>processed/fulcrum</code> and <code>processed/sanger</code> directories hold <code>easyfulcrum</code> function outputs.<br> The <code>reports</code> directory holds <code>easyfulcrum</code> function outputs. These outputs will be generated by the user processing script(s) saved in the <code>scripts</code> directory.<br></p>
<p>In a normal <code>easyfulcrum</code> run, the user would add collection <code>.csv</code> and <code>.jpg</code> files into the appropriate locations after running this function.</p>
</div>
<div id="reading-processing-and-joining-fulcrum-results" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-processing-and-joining-fulcrum-results" class="anchor"></a>Reading, processing, and joining Fulcrum results:</h3>
<p>The first group of functions cleans the results from the Fulcrum <code>.csv</code> files.<br></p>
<div id="readfulcrum" class="section level4">
<h4 class="hasAnchor">
<a href="#readfulcrum" class="anchor"></a>readFulcrum:</h4>
<p><code>readFulcrum</code> takes a <code>dir</code> argument that specifies the directory to read in Fulcrum <code>.csv</code> files. This will be useful throughout the package.</p>
<p>As this depends on a users personal subfolder structure, this function is commented out (<code>raw_data1</code> object is automatically loaded as the example raw fulcrum data).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dir1</span> <span class="op">&lt;-</span> <span class="st">"~/Desktop/2020JanuaryHawaii"</span>
<span class="co"># raw_fulc1 &lt;- readFulcrum(dir = dir1)</span>
<span class="va">raw_fulc1</span> <span class="op">&lt;-</span> <span class="fu">easyfulcrum</span><span class="fu">::</span><span class="va">raw_fulc1</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">raw_fulc1</span><span class="op">)</span>
<span class="co">#&gt; [1] "nematode_field_sampling_sample_photo"</span>
<span class="co">#&gt; [2] "nematode_field_sampling"             </span>
<span class="co">#&gt; [3] "nematode_isolation_photos"           </span>
<span class="co">#&gt; [4] "nematode_isolation_s_labeled_plates" </span>
<span class="co">#&gt; [5] "nematode_isolation"</span></code></pre></div>
</div>
<div id="procfulcrum" class="section level4">
<h4 class="hasAnchor">
<a href="#procfulcrum" class="anchor"></a>procFulcrum:</h4>
<p><code>procFulcrum</code> processes individual data frames and adds flags for unexpected data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proc_fulc1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/procFulcrum.html">procFulcrum</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">raw_fulc1</span><span class="op">)</span>
<span class="co">#&gt; Processing nematode_field_sampling</span>
<span class="co">#&gt; Processing nematode_field_sampling_sample_photo</span>
<span class="co">#&gt; Processing nematode_isolation</span>
<span class="co">#&gt; Processing nematode_isolation_s_labeled_plates</span>
<span class="co">#&gt; Processing nematode_isolation_photos</span></code></pre></div>
</div>
<div id="checktemperatures" class="section level4">
<h4 class="hasAnchor">
<a href="#checktemperatures" class="anchor"></a>checkTemperatures:</h4>
<p><code>checkTemperatures</code> identifies flags in three temperature variables. Setting the <code>return_flags</code> option to <code>TRUE</code> will return a list of three data frames that pulls only the rows where each of the three flag types appear. The function automatically prints the rows where the flags exist.<br></p>
<p><code>procFulcrum</code> function assumes that, when <code>raw_substrate_temperature</code> or <code>raw_ambient_temperature</code> temperatures are above 40 degrees, the temperatures were mistakenly input as Fahrenheit rather than Celsius, and converts these values to Celsius. It will also notice when both <code>raw_ambient_temperature</code> and <code>raw_ambient_humidity</code> get stuck on the same value for 5 or more measurements in a row. These are the three flags returned above.<br></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flag1.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkTemperatures.html">checkTemperatures</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">proc_fulc1</span>, return_flags <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking substrate temperature</span>
<span class="co">#&gt; [1] "There are 1 rows with flagged substrate temperature:"</span>
<span class="co">#&gt;                             fulcrum_id raw_substrate_temperature</span>
<span class="co">#&gt; 1 a7db618d-44cc-4b4a-bc67-871306029274                      65.3</span>
<span class="co">#&gt;   proc_substrate_temperature</span>
<span class="co">#&gt; 1                       18.5</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking ambient temperature</span>
<span class="co">#&gt; [1] "There are 1 rows with flagged ambient temperature:"</span>
<span class="co">#&gt;                             fulcrum_id raw_ambient_temperature</span>
<span class="co">#&gt; 1 b1f20ae4-c5c2-426f-894a-e1f46c2fa693                    71.2</span>
<span class="co">#&gt;   proc_ambient_temperature</span>
<span class="co">#&gt; 1                     21.8</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking ambient run temperature</span>
<span class="co">#&gt; [1] "There are 5 rows with flagged ambient run temperature:"</span>
<span class="co">#&gt;                              fulcrum_id c_label raw_ambient_temperature</span>
<span class="co">#&gt; 1  854b40c4-ac6a-47d1-a737-cf611aa94268  C-5115                    16.9</span>
<span class="co">#&gt; 2  de2ae06e-cfff-44bc-869e-8dd9205a65c6  C-5139                    23.3</span>
<span class="co">#&gt; 3  b1f20ae4-c5c2-426f-894a-e1f46c2fa693  C-5121                    71.2</span>
<span class="co">#&gt; 4  0a049170-e63c-4545-8b02-04bc08a5ea54  C-5117                    16.7</span>
<span class="co">#&gt; 5  34848a83-5951-4136-b632-d39aa85c76d3  C-5120                    17.4</span>
<span class="co">#&gt; 6  dda77efe-d73c-48e9-aefb-b508e613256b  C-5122                    17.4</span>
<span class="co">#&gt; 7  93de14a0-40ab-4793-8614-ab1512ab158c  C-5118                    17.4</span>
<span class="co">#&gt; 8  216cb71a-6470-46eb-950d-366ac3180498  C-5123                    17.4</span>
<span class="co">#&gt; 9  920a6a56-7a29-47f4-afce-a5f83787d639  C-5131                    17.4</span>
<span class="co">#&gt; 10 6b25c113-4bb6-4bc5-9473-eca1f8075d10  C-5132                    17.4</span>
<span class="co">#&gt; 11 e38a295e-7595-4227-887b-74e1514c71bc  C-5133                    17.4</span>
<span class="co">#&gt; 12 a4afcda9-2d13-4f7f-991a-2299cb3c5a3c  C-5135                    15.8</span>
<span class="co">#&gt; 13 c0e68d42-5180-4fda-88e8-7721bbc8c96d  C-5134                    15.3</span>
<span class="co">#&gt; 14 4b7c21ea-2987-49d8-88c3-951417de4965  C-5130                    15.3</span>
<span class="co">#&gt; 15 2922c799-0dd7-499c-b674-ea99c72e5ea7  C-5134                    15.3</span>
<span class="co">#&gt;    proc_ambient_temperature ambient_humidity flag_ambient_temperature_run</span>
<span class="co">#&gt; 1                      16.9             85.8                        FALSE</span>
<span class="co">#&gt; 2                      23.3             65.3                        FALSE</span>
<span class="co">#&gt; 3                      21.8             77.9                        FALSE</span>
<span class="co">#&gt; 4                      16.7             85.9                        FALSE</span>
<span class="co">#&gt; 5                      17.4             80.0                        FALSE</span>
<span class="co">#&gt; 6                      17.4             80.0                         TRUE</span>
<span class="co">#&gt; 7                      17.4             80.0                         TRUE</span>
<span class="co">#&gt; 8                      17.4             80.0                         TRUE</span>
<span class="co">#&gt; 9                      17.4             80.0                         TRUE</span>
<span class="co">#&gt; 10                     17.4             80.0                         TRUE</span>
<span class="co">#&gt; 11                     17.4             87.0                        FALSE</span>
<span class="co">#&gt; 12                     15.8             90.2                        FALSE</span>
<span class="co">#&gt; 13                     15.3             91.2                        FALSE</span>
<span class="co">#&gt; 14                     15.3             90.2                        FALSE</span>
<span class="co">#&gt; 15                     15.3             87.8                        FALSE</span>
<span class="co">#&gt;    collection_local_time collection_datetime_UTC</span>
<span class="co">#&gt; 1             42660 secs     2020-01-19 21:51:26</span>
<span class="co">#&gt; 2             43080 secs     2020-01-19 21:58:02</span>
<span class="co">#&gt; 3             43140 secs     2020-01-19 21:59:02</span>
<span class="co">#&gt; 4             43320 secs     2020-01-19 22:02:01</span>
<span class="co">#&gt; 5             44100 secs     2020-01-19 22:15:15</span>
<span class="co">#&gt; 6             45780 secs     2020-01-19 22:43:04</span>
<span class="co">#&gt; 7             45960 secs     2020-01-19 22:46:11</span>
<span class="co">#&gt; 8             46440 secs     2020-01-19 22:54:01</span>
<span class="co">#&gt; 9             38520 secs     2020-01-22 20:42:55</span>
<span class="co">#&gt; 10            39000 secs     2020-01-22 20:50:48</span>
<span class="co">#&gt; 11            40320 secs     2020-01-22 21:12:29</span>
<span class="co">#&gt; 12            45420 secs     2020-01-22 22:37:28</span>
<span class="co">#&gt; 13            45540 secs     2020-01-22 22:39:10</span>
<span class="co">#&gt; 14            47400 secs     2020-01-22 23:10:25</span>
<span class="co">#&gt; 15            51780 secs     2020-01-23 00:23:02</span></code></pre></div>
</div>
<div id="fixtemperatures" class="section level4">
<h4 class="hasAnchor">
<a href="#fixtemperatures" class="anchor"></a>fixTemperatures:</h4>
<p><code>fixTemperatures</code> takes a) <code>fulcrum_id</code>s that need to be reverted <em>back to their original values</em> (if readings above 40 degrees were truly in Celsius) for both substrate temperatures (<code>substrate_temperature_ids</code>) and ambient temperatures (<code>ambient_temperature_ids</code>) as well as b) <code>fulcrum_id</code>s for which humidity and temperature readings need to be set to NA due to a stuck measurement device (<code>ambient_temperature_run_ids</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proc_fulc1_clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fixTemperatures.html">fixTemperatures</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">proc_fulc1</span>,
                                  substrate_temperature_ids <span class="op">=</span> <span class="st">"a7db618d-44cc-4b4a-bc67-871306029274"</span>,
                                  ambient_temperature_ids <span class="op">=</span> <span class="st">"b1f20ae4-c5c2-426f-894a-e1f46c2fa693"</span>,
                                  ambient_temperature_run_ids<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dda77efe-d73c-48e9-aefb-b508e613256b"</span>,
                                                                 <span class="st">"93de14a0-40ab-4793-8614-ab1512ab158c"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The flag variables will be maintained, so rerunning <code>checkTemperatures</code> on the cleaned processed fulcrum data to ensure that corrections have been implemented as desired.</p>
</div>
<div id="joinfulcrum" class="section level4">
<h4 class="hasAnchor">
<a href="#joinfulcrum" class="anchor"></a>joinFulcrum:</h4>
<p><code>joinFulcrum</code> joins the Fulcrum dataframes. The function works to first join the processed nematode field sampling dataframe to the processed nematode isolation dataframe via unique collection labels (<code>c_label</code>). Following this join, it selects a “best” photo for each unique C-label based on the existence of a matching photo id in the processed nematode field sampling sample photo dataframe, before joining the aggregated data to this dataframe as well. Finally, the complete merge of the dataframes is achieved when the processed nematode isolation S-labeled plates is joined to this large dataframe on the basis of isolation id (<code>s_label</code>).<br></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">join_fulc1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/joinFulcrum.html">joinFulcrum</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">proc_fulc1</span><span class="op">)</span>
<span class="co">#&gt; Attempting to join:</span>
<span class="co">#&gt; proc_fulc1$nematode_field_sampling_proc</span>
<span class="co">#&gt; proc_fulc1$nematode_field_sampling_sample_photo_proc</span>
<span class="co">#&gt; proc_fulc1$nematode_isolation_proc</span>
<span class="co">#&gt; proc_fulc1$nematode_isolation_s_labeled_plates_proc</span>
<span class="co">#&gt; proc_fulc1$nematode_isolation_photos_proc</span>
<span class="co">#&gt; Complete fulcrum data detected, joining all data.</span></code></pre></div>
</div>
<div id="checkjoin" class="section level4">
<h4 class="hasAnchor">
<a href="#checkjoin" class="anchor"></a>checkJoin:</h4>
<p><code>checkJoin</code> conducts 10 different checks for flags in the joined Fulcrum data frame regarding extreme temperatures and altitude values, missing, improper, and/or duplicated C-labels, missing, and/or duplicated isolation records corresponding to these C-labels, and finally unusual sample photo numbers. The <code>return_flags</code> option is the same as in <code>checkTemperatures</code>, and the function automatically prints the rows where the flags exist. If desired, the user can manually edit values or correct mistakes in the underlying data based on these flags and re-run the pipeline again.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flag1.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkJoin.html">checkJoin</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">join_fulc1</span>, return_flags <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking data classes</span>
<span class="co">#&gt; [1] "There are 1 improperly classified variables"</span>
<span class="co">#&gt; [1] "isolation_date_UTC"</span>
<span class="co">#&gt; [1] "Improperly classified variables may require manipulation after read-in"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking duplicated c labels</span>
<span class="co">#&gt; [1] "There are 2 rows with duplicated c labels, these c labels are:"</span>
<span class="co">#&gt; [1] "C-5134" "C-5134"</span>
<span class="co">#&gt; [1] "Duplicated c labels are found in nematode_field_sampling.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking unusual sample photo number</span>
<span class="co">#&gt; [1] "There are 6 rows with unusual sample photo numbers, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5121" "C-5121" "C-5121" "C-5121" "C-5121" "C-5121"</span>
<span class="co">#&gt; [1] "Unusual sample photo number are found in nematode_field_sampling.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking duplicated isolation for c label</span>
<span class="co">#&gt; [1] "There are 4 rows with duplicated isolation for c label, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5130" "C-5130" "C-5130" "C-5130"</span>
<span class="co">#&gt; [1] "Duplicated isolation for c label are found in nematode_isolation.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking missing isolation records</span>
<span class="co">#&gt; [1] "There are 4 rows with missing isolation records, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5130" "C-5130" "C-5130" "C-5133"</span>
<span class="co">#&gt; [1] "Missing isolation records are found in nematode_isolation.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking extreme substrate temperatures</span>
<span class="co">#&gt; [1] "There are 6 rows with extreme substrate temperatures, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5121" "C-5121" "C-5121" "C-5121" "C-5121" "C-5121"</span>
<span class="co">#&gt; [1] "Extreme substrate temperatures are found in nematode_field_sampling.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking extreme ambient temperatures</span>
<span class="co">#&gt; [1] "There are 3 rows with extreme ambient temperatures, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5126" "C-5126" "C-5126"</span>
<span class="co">#&gt; [1] "Extreme ambient temperatures are found in nematode_field_sampling.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking extreme collection altitude</span>
<span class="co">#&gt; [1] "There are 1 rows with extreme collection altitudes, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5134"</span>
<span class="co">#&gt; [1] "Extreme collection altitudes are found in nematode_field_sampling.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking missing s labels</span>
<span class="co">#&gt; [1] "There are 1 rows with missing s labels, their c labels are:"</span>
<span class="co">#&gt; [1] "C-5115"</span>
<span class="co">#&gt; [1] "Missing s labels are found in nematode_isolation_s_labeled_plates.csv"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking duplicated s labels</span>
<span class="co">#&gt; [1] "There are 2 rows with duplicated s labels, their s labels are:"</span>
<span class="co">#&gt; [1] "S-12776" "S-12776"</span>
<span class="co">#&gt; [1] "Duplicated s labels are found in nematode_isolation_s_labeled_plates.csv"</span></code></pre></div>
</div>
<div id="annotatefulcrum" class="section level4">
<h4 class="hasAnchor">
<a href="#annotatefulcrum" class="anchor"></a>annotateFulcrum:</h4>
<p><code>annnotateFulcrum</code> adds spatial information to the joined Fulcrum data frame, noting if sample collections were collected on specific islands, trails, and/or locations. Examples islands, trails, and/or locations are automatically loaded with the package, but a user can specify manually made <code>.csv</code> files and place them in <em>data/raw/annotate</em>, specifying the base directory as <code>dir</code> in <code>annotateFulcrum</code> will override the example files.<br></p>
<p>The <code>hawaii_islands</code> and <code>hawaii_locations</code> dataframes are composed of simple latitude and longitude starts and ends to create a bounding box, and <code>hawaii_trails</code> is composed of a character list of geojson polygon points from geojson output of that can be created on a <a href="https://boundingbox.klokantech.com/">bounding box online tool</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">anno_fulc1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotateFulcrum.html">annotateFulcrum</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">join_fulc1</span>, dir <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="reading-processing-and-joining-genotyping-google-sheet" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-processing-and-joining-genotyping-google-sheet" class="anchor"></a>Reading, processing, and joining genotyping google sheet:</h3>
<p>This second group of functions operates to clean the results from the collection specific Google Sheet of genotyping results.<br></p>
<p>Details on how to create genotyping Google Sheets can be found in the <a href="https://docs.google.com/document/d/1aPgPDYobpxzr0BQYgvUsjPWnAGXZmCSQKQ4sDGBXuYM/edit">Nematode Collection Protocol</a>, look for “wild_isolate_genotyping_template”. A template is <a href="https://docs.google.com/spreadsheets/d/1pvdpujuJDTYaMwjjDlivjAnO0MrQuDguU2kRLs9eemQ/edit#gid=0">linked</a><br></p>
<div id="readgenotypes" class="section level4">
<h4 class="hasAnchor">
<a href="#readgenotypes" class="anchor"></a>readGenotypes:</h4>
<p><code>readGenotypes</code> reads in genotyping data from a Google Sheet with requisite <code>gsKey</code>. The <code>col_types</code> variable will specify of what column types the data should be.<br></p>
<p>For more details on reading in genotyping sheets, look into the <a href="https://googlesheets4.tidyverse.org/articles/articles/read-sheets.html">googlesheets4</a> package (which underlies this function), as well as further information on how to <a href="https://googlesheets4.tidyverse.org/reference/range_read.html#column-specification">specify</a> the <code>col_types</code> if needed.<br></p>
<p>As this depends on a users connection to the appropriate google doc, this function is commented out (<code>raw_geno1</code> object is automatically loaded as the example raw geno data).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># raw_geno1 &lt;- readGenotypes(gsKey = c("1_6u4sk_Zj-Hm5d_058Lg8WYWLe7BZHGTWxXcH6EsDUI"))</span>
<span class="va">raw_geno1</span> <span class="op">&lt;-</span> <span class="fu">easyfulcrum</span><span class="fu">::</span><span class="va">raw_geno1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">raw_geno1</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 38</span>
<span class="co">#&gt;   project_id   shipment_sent_d… shipment_receive… shipment_number shipment_color</span>
<span class="co">#&gt;   &lt;chr&gt;        &lt;date&gt;           &lt;date&gt;                      &lt;dbl&gt; &lt;chr&gt;         </span>
<span class="co">#&gt; 1 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; 2 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; 3 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; 4 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; 5 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; 6 2020January… NA               NA                              1 &lt;NA&gt;          </span>
<span class="co">#&gt; # … with 33 more variables: isolation_box_number &lt;dbl&gt;, s_label &lt;chr&gt;,</span>
<span class="co">#&gt; #   s_label_repeat_error &lt;dbl&gt;, proliferation_48 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   proliferation_168 &lt;dbl&gt;, proliferating &lt;dbl&gt;, strip_tube_number &lt;dbl&gt;,</span>
<span class="co">#&gt; #   strip_tube_position &lt;dbl&gt;, lysis_date &lt;chr&gt;, lysis_notes &lt;chr&gt;,</span>
<span class="co">#&gt; #   ssu_pcr_date &lt;date&gt;, its2_pcr_date &lt;date&gt;, pcr_plate_number &lt;dbl&gt;,</span>
<span class="co">#&gt; #   pcr_well &lt;chr&gt;, gel_number &lt;dbl&gt;, gel_position &lt;chr&gt;,</span>
<span class="co">#&gt; #   pcr_product_ssu &lt;dbl&gt;, positive_ctrl_ssu &lt;dbl&gt;, neg_ctrl_ssu &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ssu_fail_redo &lt;dbl&gt;, pcr_product_its2 &lt;dbl&gt;, positive_ctrl_its2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   neg_ctrl_ist2 &lt;dbl&gt;, general_notes &lt;chr&gt;, mclab_sequencing_plate &lt;dbl&gt;,</span>
<span class="co">#&gt; #   mclab_sequencing_well &lt;chr&gt;, manual_blast_notes &lt;chr&gt;, species_id &lt;chr&gt;,</span>
<span class="co">#&gt; #   possible_new_caeno_sp &lt;dbl&gt;, make_strain_name &lt;dbl&gt;,</span>
<span class="co">#&gt; #   reason_strain_not_named &lt;chr&gt;, strain_name &lt;chr&gt;, pair_name &lt;chr&gt;</span></code></pre></div>
</div>
<div id="checkgenotypes" class="section level4">
<h4 class="hasAnchor">
<a href="#checkgenotypes" class="anchor"></a>checkGenotypes:</h4>
<p><code>checkGenotypes</code> both processes the genotyping data (adds flags) and returns info on those flags if desired, regarding isolations missing, improper, and/or duplicated S-labels, and other checks such as if species ID, strain name, proliferation label and ITS2 genotype are missing and expected based on data in the genotyping sheet. Setting the <code>return_geno</code> option to <code>TRUE</code> and the <code>return_flags</code> option to <code>FALSE</code> will return the processed genotyping data and print information on the flags. Setting the <code>return_geno</code> option to <code>FALSE</code> and the <code>return_flags</code> option to <code>TRUE</code> will return a list of data frames that detail the rows where the flags appear. Both of these cannot be <code>TRUE</code> at the same time.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proc_geno1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkGenotypes.html">checkGenotypes</a></span><span class="op">(</span>geno_data <span class="op">=</span> <span class="va">raw_geno1</span>, fulc_data <span class="op">=</span> <span class="va">anno_fulc1</span>, 
                                  return_geno <span class="op">=</span> <span class="cn">TRUE</span>, return_flags <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking data classes</span>
<span class="co">#&gt; [1] "There are 1 improperly classified variables"</span>
<span class="co">#&gt; [1] "lysis_date"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking s labels</span>
<span class="co">#&gt; [1] "There are 1 rows with missing s labels, these s labels are:"</span>
<span class="co">#&gt; [1] NA</span>
<span class="co">#&gt; [1] "There are 2 rows with duplicated s labels, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12664" "S-12664"</span>
<span class="co">#&gt; [1] "There are 1 rows with unusual s labels, these s labels are:"</span>
<span class="co">#&gt; [1] "R-12666"</span>
<span class="co">#&gt; [1] "There are 3 rows with s labels not found in the Fulcrum data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12652" "R-12666" "S-12688"</span>
<span class="co">#&gt; [1] "There are 3 s labels in the Fulcrum data but not in the genotyping data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12666" "S-12662" "S-12663"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking genotyping process</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected proliferation data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12668"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected its2 genotype, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12652"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected species_id, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12668"</span>
<span class="co">#&gt; [1] "There are 1 rows with unusual target species names, these names are:"</span>
<span class="co">#&gt; [1] "O. myriophilus"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected strain_name, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12775"</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flag1.3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkGenotypes.html">checkGenotypes</a></span><span class="op">(</span>geno_data <span class="op">=</span> <span class="va">raw_geno1</span>, fulc_data <span class="op">=</span> <span class="va">anno_fulc1</span>, 
                          return_geno <span class="op">=</span> <span class="cn">FALSE</span>, return_flags <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking data classes</span>
<span class="co">#&gt; [1] "There are 1 improperly classified variables"</span>
<span class="co">#&gt; [1] "lysis_date"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking s labels</span>
<span class="co">#&gt; [1] "There are 1 rows with missing s labels, these s labels are:"</span>
<span class="co">#&gt; [1] NA</span>
<span class="co">#&gt; [1] "There are 2 rows with duplicated s labels, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12664" "S-12664"</span>
<span class="co">#&gt; [1] "There are 1 rows with unusual s labels, these s labels are:"</span>
<span class="co">#&gt; [1] "R-12666"</span>
<span class="co">#&gt; [1] "There are 3 rows with s labels not found in the Fulcrum data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12652" "R-12666" "S-12688"</span>
<span class="co">#&gt; [1] "There are 3 s labels in the Fulcrum data but not in the genotyping data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12666" "S-12662" "S-12663"</span>
<span class="co">#&gt; &gt;&gt;&gt; Checking genotyping process</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected proliferation data, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12668"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected its2 genotype, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12652"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected species_id, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12668"</span>
<span class="co">#&gt; [1] "There are 1 rows with unusual target species names, these names are:"</span>
<span class="co">#&gt; [1] "O. myriophilus"</span>
<span class="co">#&gt; [1] "There are 1 rows missing expected strain_name, these s labels are:"</span>
<span class="co">#&gt; [1] "S-12775"</span></code></pre></div>
</div>
<div id="joingenofulc" class="section level4">
<h4 class="hasAnchor">
<a href="#joingenofulc" class="anchor"></a>joinGenoFulc:</h4>
<p><code>joinGenoFulc</code> will join the joined Fulcrum data frame with the genotyping information. This function will also save the processed genotyping information in <em>data/processed/genotypes</em> if desired.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">join_genofulc1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/joinGenoFulc.html">joinGenoFulc</a></span><span class="op">(</span>geno <span class="op">=</span> <span class="va">proc_geno1</span>, fulc <span class="op">=</span> <span class="va">anno_fulc1</span>, dir <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "s_label"</span></code></pre></div>
</div>
</div>
<div id="reading-resizing-and-joining-collection-images" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-resizing-and-joining-collection-images" class="anchor"></a>Reading, resizing, and joining collection images:</h3>
<p>The final function processes and resizes images, adding details to a final dataframe.<br></p>
<div id="procphotos" class="section level4">
<h4 class="hasAnchor">
<a href="#procphotos" class="anchor"></a>procPhotos:</h4>
<p><code>procPhotos</code> processes the raw sample photos and adds information regarding them to the final data frame of data, which is also saved in <em>data/processed/fulcrum</em>. Setting the <code>CeNDR</code> option to <code>TRUE</code> will place photos of samples meeting CeNDR criteria into an appropriate subfolder.</p>
<p>As this depends on a users personal subfolder structure (<code>dir1</code>) and it containing the proper images, this function is commented out (<code>final_data1</code> object is automatically loaded as the example final merged data). Because <code>procPhotos</code> is not executed, images are not moved or resized for the purposes of the vignette build.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># final_data1 &lt;- procPhotos(dir = dir1, data = join_genofulc1,</span>
<span class="co">#                           max_dim = 500, overwrite = T, </span>
<span class="co">#                           CeNDR = TRUE)</span>
<span class="va">final_data1</span> <span class="op">&lt;-</span> <span class="fu">easyfulcrum</span><span class="fu">::</span><span class="va">final_data1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">final_data1</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 95</span>
<span class="co">#&gt;   project           c_label s_label species_id strain_name collection_by        </span>
<span class="co">#&gt;   &lt;chr&gt;             &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;                </span>
<span class="co">#&gt; 1 2020JanuaryHawaii C-5115  S-12695 &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; 2 2020JanuaryHawaii C-5115  &lt;NA&gt;    &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; 3 2020JanuaryHawaii C-5115  S-12706 &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; 4 2020JanuaryHawaii C-5115  S-12697 &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; 5 2020JanuaryHawaii C-5115  S-12776 &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; 6 2020JanuaryHawaii C-5115  S-12776 &lt;NA&gt;       &lt;NA&gt;        dec@u.northwestern.e…</span>
<span class="co">#&gt; # … with 89 more variables: collection_datetime_UTC &lt;dttm&gt;,</span>
<span class="co">#&gt; #   collection_date_UTC &lt;date&gt;, collection_local_time &lt;drtn&gt;,</span>
<span class="co">#&gt; #   collection_fulcrum_latitude &lt;dbl&gt;, collection_fulcrum_longitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   exif_gps_latitude &lt;dbl&gt;, exif_gps_longitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   collection_latitude &lt;dbl&gt;, collection_longitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   collection_lat_long_method &lt;chr&gt;, collection_lat_long_method_diff &lt;dbl&gt;,</span>
<span class="co">#&gt; #   fulcrum_altitude &lt;dbl&gt;, exif_gps_altitude &lt;dbl&gt;, collection_altitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   collection_altitude_method &lt;chr&gt;, collection_location &lt;chr&gt;,</span>
<span class="co">#&gt; #   collection_island &lt;chr&gt;, collection_trail &lt;chr&gt;, landscape &lt;chr&gt;,</span>
<span class="co">#&gt; #   sky_view &lt;chr&gt;, ambient_humidity &lt;dbl&gt;, substrate &lt;chr&gt;,</span>
<span class="co">#&gt; #   substrate_notes &lt;chr&gt;, substrate_other &lt;lgl&gt;,</span>
<span class="co">#&gt; #   raw_ambient_temperature &lt;dbl&gt;, proc_ambient_temperature &lt;dbl&gt;,</span>
<span class="co">#&gt; #   raw_substrate_temperature &lt;dbl&gt;, proc_substrate_temperature &lt;dbl&gt;,</span>
<span class="co">#&gt; #   gridsect &lt;chr&gt;, gridsect_index &lt;lgl&gt;, gridsect_radius &lt;lgl&gt;,</span>
<span class="co">#&gt; #   grid_sect_direction &lt;lgl&gt;, sample_photo1 &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo1_raw_file_name &lt;chr&gt;, sample_photo1_processed_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo1_hash &lt;chr&gt;, sample_photo1_resized_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo1_resized_hash &lt;chr&gt;, sample_photo2 &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo2_raw_file_name &lt;chr&gt;, sample_photo2_processed_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo2_hash &lt;chr&gt;, sample_photo2_resized_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo2_resized_hash &lt;chr&gt;, sample_photo3 &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo3_raw_file_name &lt;chr&gt;, sample_photo3_processed_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo3_hash &lt;chr&gt;, sample_photo3_resized_file_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo3_resized_hash &lt;chr&gt;, best_exif_dop_photo &lt;chr&gt;,</span>
<span class="co">#&gt; #   best_sample_photo_caption &lt;lgl&gt;, isolation_by &lt;chr&gt;,</span>
<span class="co">#&gt; #   isolation_datetime_UTC &lt;chr&gt;, isolation_date_UTC &lt;chr&gt;,</span>
<span class="co">#&gt; #   isolation_local_time &lt;drtn&gt;, isolation_latitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   isolation_longitude &lt;dbl&gt;, worms_on_sample &lt;chr&gt;,</span>
<span class="co">#&gt; #   approximate_number_of_worms &lt;chr&gt;, shipment_sent_date &lt;date&gt;,</span>
<span class="co">#&gt; #   shipment_received_date &lt;date&gt;, proliferation_48 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   proliferation_168 &lt;dbl&gt;, proliferating &lt;dbl&gt;, lysis_date &lt;chr&gt;,</span>
<span class="co">#&gt; #   pcr_product_its2 &lt;dbl&gt;, pcr_product_ssu &lt;dbl&gt;, general_notes &lt;chr&gt;,</span>
<span class="co">#&gt; #   manual_blast_notes &lt;chr&gt;, possible_new_caeno_sp &lt;dbl&gt;,</span>
<span class="co">#&gt; #   make_strain_name &lt;dbl&gt;, reason_strain_not_named &lt;chr&gt;,</span>
<span class="co">#&gt; #   flag_ambient_temperature &lt;lgl&gt;, flag_ambient_temperature_run &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_substrate_temperature &lt;lgl&gt;, flag_unusual_sample_photo_num &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_duplicated_c_label_field_sampling &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_duplicated_isolation_for_c_label &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_duplicated_s_label_isolation_s_labeled_plates &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_missing_s_label_isolation_s_labeled_plates &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_missing_isolation_record &lt;lgl&gt;, flag_unusual_s_label_genotyping &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_missing_s_label_genotyping &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_duplicated_s_label_genotyping &lt;lgl&gt;,</span>
<span class="co">#&gt; #   flag_unusual_target_species_name &lt;lgl&gt;, sample_photo1_processed_url &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_photo2_processed_url &lt;chr&gt;, sample_photo3_processed_url &lt;lgl&gt;</span></code></pre></div>
</div>
</div>
<div id="generating-summary-and-output-files" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-summary-and-output-files" class="anchor"></a>Generating summary and output files:</h3>
<p>We include two functions for generating sample output from <code>final_data1</code>. This final dataframe can otherwise be used as needed by the user. As these depend on a users personal subfolder structure, they are commented out.</p>
<div id="makespsheet" class="section level4">
<h4 class="hasAnchor">
<a href="#makespsheet" class="anchor"></a>makeSpSheet:</h4>
<p><code>makeSpSheet</code> generates a species specific <code>.csv</code> file for the species of interest (<code>target_sp</code>) and writes it to the <code>/reports</code> subdirectory. This function simplifies the output of the final dataframe, pulling variables of particular interest for a user specified species of interest. In addition, this function standardizes an output dataframe to meet the specifications of submitting to the CaeNDR.<br></p>
<p>It also returns a dataframe with flags for these select samples, and prints a description of these flags.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># flag1.4 &lt;- makeSpSheet(data = final_data1,</span>
<span class="co">#                     target_sp = "Caenorhabditis briggsae", dir = dir1)</span></code></pre></div>
</div>
<div id="samplereport" class="section level4">
<h4 class="hasAnchor">
<a href="#samplereport" class="anchor"></a>sampleReport:</h4>
<p><code>sampleReport</code> is a markdown file that will generate a simple description of the data in the project based on <code>final_data1</code>. The file will generate output to summarize collection project metadata (such as who conducted the respective processes and on what dates they were completed), to generate summary tables of collection and isolation data, to produce visual maps of where the collections in the project were acquired, and finally to graph basic box and whisker plots of substrate temperature, ambient temperature, humidity, and elevation for the project.</p>
<p>Please feel free to edit this file as you require, the first command will copy the file .Rmd file into <code>/scripts</code> sub-directory. We suggest specifying a location for output of the .html file, such as the <code>/reports</code> sub-directory.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># file.copy("../R/sampleReport.Rmd", paste(dir1,"scripts",sep = "/"))</span>
<span class="co"># </span>
<span class="co"># rmarkdown::render(paste(dir1,"scripts","sampleReport.Rmd",sep = "/"),</span>
<span class="co">#                   output_dir = paste(dir1,"reports",sep = "/"),</span>
<span class="co">#                   output_format   = "html_document")</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tim Crombie, Daniel Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
